### DISCUSSION TITLE
[Safari] [Shortcuts] - Open YouTube links automatically in FreeTube from Safari

### DISCUSSION THREAD

**Discussion by OblackatO** [FreeTubeApp/FreeTube#4390] [General]
After reading the [FAQ](https://docs.freetubeapp.io/faq/) and the [Browser Extensions](https://docs.freetubeapp.io/usage/browser-extension/) sections, I didn't find a direct way to automatically open Youtube links from Safari. 

One of the extensions mentioned, has a MacOS [version](https://apps.apple.com/us/app/privacy-redirect/id1578144015) which currently costs 1,99€. 

It is quite easy to open YouTube links from Safari in FreeTube, using standard functionalities of [shortcusts](https://support.apple.com/en-en/guide/shortcuts-mac/apdf22b0444c/mac), which comes pre-installed with macOS and iOS. 

I also didn't find any issue about this. Hence this discussion. Would it be possible to add documentation about this topic? 

I can provide all the necessary information.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
This is a known bug, it was reported in https://github.com/FreeTubeApp/FreeTube/discussions/2747 and converted in an issue https://github.com/FreeTubeApp/FreeTube/issues/3257

**Reply by OblackatO** [reply]
No... It is not about the bug. It works without an extension, just using standard features of macOS. There's 0 need for an extension to do this.

**Reply by OblackatO** [reply]
I will document this elsewhere. Thanks for the feedback.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Apologies not sure i understand the question then. Also FYI the browser extension u linked that has a macOS version is an unofficial version of Privacy Redirect

**Reply by OblackatO** [reply]
I meant we don't need an extension to automatically open videos from Youtube on FreeTube, in macOS with Safari. 

The links  in my description were to quickly show the searches I did before opening this discussion. Let me make a small video and take some screenshots. I will be more clear :)

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Ah that would be awesome for macOS users to use built in shortcuts and to not use an extension

**Comment by OblackatO** [comment]
# How to automatically open Youtube links in FreeTube using macOS shortcuts
1. Open the Shortcuts App:
2. Click on the “+” button to create a new shortcut with the following actions *(please ignore all screenshots in PT-PT. I will translate this if necessary)*

![Captura de ecrã 2023-11-26, às 20 17 23](https://github.com/FreeTubeApp/FreeTube/assets/25124288/e74a0845-3b0f-4f62-b7ec-be77caefc09a)

The script basically gets the URL from the current opened tab in Safari and checks if *youtube.com* is in the URL. If not, the script checks the clipboard and checks again if *youtube.com* is in the clipboard. If both cases fail, an error message is displayed: 

  <img width="356" alt="Captura de ecrã 2023-11-26, às 20 25 27" src="https://github.com/FreeTubeApp/FreeTube/assets/25124288/a1e638e7-04bb-4292-a480-3576c686987d">

3. Open System Preferences:
- Go to the Apple menu at the top left corner of your screen and select "System Preferences."

4. Navigate to Keyboard Settings:
- Click on "Keyboard" within System Preferences.

5. Open Shortcuts Tab:
- In the Keyboard settings window, go to the "Shortcuts" tab.

6. Select Services:
- On the left-hand side, select "Services" from the list. This section contains all the services and shortcuts available on your Mac.
- Set some keyboard shortcut to launch the shortcut on-demand... In the screenshot below I chose shift+command+k

<img width="689" alt="Captura de ecrã 2023-11-26, às 20 29 28" src="https://github.com/FreeTubeApp/FreeTube/assets/25124288/3870ac49-0fce-4260-a818-f5fa9bfff69b">


The shortcut should be ready to use... 

In the following video, I firstly show FreeTube. 
Then, I moved the screen to the right and an Youtube video is already running. Even though you don't see it, I run my shortcut pressing *shift+command+k*. When returning to FreeTube, the video is already loading. Note that moving to FreeTube wasn't automatically done, but probably can be as well. Note that I DID NOT copy the link from YouTube. That was automatically gathered by the shortcut. 

https://mega.nz/file/HupWXLpY#9W6_Q2w9KnNUdsIVxQ5--TXaI5xwkql9G9QAX6v9Q6U

I couldn't upload the video here, because it is bigger than 10MB. If you can't watch it in the browser and prefer not to download it, that's ok. It is just for the sake of completeness.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Compressed the video with HandBrake, its now 2,46 MB :)


https://github.com/FreeTubeApp/FreeTube/assets/73130443/123b2a12-cd32-43d8-ba76-354e9db2b7d7

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
@PikachuEXE could u test how this works for u

**Reply by OblackatO** [reply]
Great ! Thanks.

I forgot to mention that even if the description is long, the setup shouldn’t take more than 10-15min if the users are familiar with macOS and shortcuts in general.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
If this has been tested this should definitely be included on https://docs.freetubeapp.io/usage/

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Hi @azlekov maybe this could be a workaround for your previously reported issue in https://github.com/FreeTubeApp/FreeTube/discussions/2747

**Reply by PikachuEXE** [reply]
I really need translation of that shortcuts app screenshot
Got too many options to guess which one to use (I have never used it)
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/726a3822-e851-41c2-8bb6-f371de3a1e50)

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
https://support.apple.com/guide/shortcuts-mac/run-a-shortcut-from-the-app-apd5ba077760/7.0/mac/14.0

Maybe this helps?

**Reply by PikachuEXE** [reply]
I am talking about step 2 `Click on the “+” button to create a new shortcut with the following actions (please ignore all screenshots in PT-PT. I will translate this if necessary)`
No idea how to config if I can't read it
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/251f3626-56e1-4609-9a04-9fd249cffa89)

**Reply by OblackatO** [reply]
You got it. 

![Captura de ecrã 2023-11-27, às 06 09 46](https://github.com/FreeTubeApp/FreeTube/assets/25124288/08c69b5e-8e45-4072-854b-cdf43ddea565)


Unrelated to this issue, but I just noticed now that it is possible to change the language settings in [specific apps only](https://osxdaily.com/2021/06/02/how-change-language-specific-apps-mac/), without changing the language of the OS. FYI..

In the parts *If Page URL contains*, you have to click upon it to select it. It is not the default option. *Get current web page from Safari*, provides several pieces of information about the web page including the URL., which is what we want.

I hope it is more clear now.

**Reply by PikachuEXE** [reply]
oops, forgot to use reply
https://github.com/FreeTubeApp/FreeTube/discussions/4390#discussioncomment-7677059

it works as expected with keyboard shortcut
It requires user as admin which some user might not be (I have no idea), might worth mentioning in doc

**Reply by OblackatO** [reply]
Nice ! 

If it is ok, I can open a documentation merge request. I would need to read how your documentation works exactly and the merge request guidelines. I will do that this week or next Friday evening. If you're in a rush feel free to do it yourselves... :) 

Could you maybe point me out some resources about the merge requests / docs you use on this project?

**Reply by PikachuEXE** [reply]
I think the repo for FT docs should be https://github.com/FreeTubeApp/FreeTube-Docs

**Comment by PikachuEXE** [comment]
I am on macOS 13.6.2 and i cannot find my created shortcut in Keyboard > Keyboard Shortcuts...

![Screenshot 2023-11-27 at 13 56 42](https://github.com/FreeTubeApp/FreeTube/assets/1018543/b30c526d-d058-43e8-81a8-1152bf79445c)

![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/7207ed72-1e16-481b-b939-266363535aa7)


Instead I need to set keyboard shortcut in the "shortcut details"
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/8033aade-ed94-47e5-b1b0-081976daec25)

**Comment by OblackatO** [comment]
That is even more practical... I will adapt the description accordingly. 

Did you face any other problems? How did it go for you?

**Comment by OblackatO** [comment]
Note to self when doing docs: FreeTube does NOT need to be opened when launching shortcut

**Comment by OblackatO** [comment]
Note to self: After the latest macOS update the keyboard key bind seems to have been delete. This might be worth mentioning. Maybe when the OS is updated some shortcuts' keybindings might be reset.

**Comment by OblackatO** [comment]
Closing. Further discussions can continue here: https://github.com/FreeTubeApp/FreeTube-Docs/pull/112
